test/issues/issues-test.js

```diff
@@ -508,4 +508,12 @@ describe("issues", function () {
             sinon.assert.calledWith(spy, secondObj);
         });
     });
+
+    describe("#1796 - cannot stub Array.prototype.sort", function () {
+        it("it should not fail with RangeError", function () {
+            var stub = sinon.stub(Array.prototype, "sort");
+            [1, 2, 3].sort();
```

Comment:
All tests should have assertions. Assert that this doesn't throw.

Reply:
Good point. I've added an extra commit. Once happy with this PR, I'll fold the commits together